### PR TITLE
Filedelete plugin refactored and fixed.

### PR DESCRIPTION
**Pull request by disaykin**
* Print information about detected deleted or changed files in filedelete_use_injector mode.
* Some code duplication removed.
* Memory leaks fixed.
* Static counter usage eliminated.
* Don't mix C++ and C style memory management.

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test.

File: src/plugins/filedelete/private.h [hunk 1]
```diff
@@ -102,8 +102,7 @@
  *                                                                         *
  ***************************************************************************/
 
-#ifndef FILEDELETE_PRIVATE_H
-#define FILEDELETE_PRIVATE_H
```
Comment (tklengyel) [inline src/plugins/filedelete/private.h]: I would prefer the #ifndef/#define style here, just for consistency.

Comment (disaykin) [reply src/plugins/filedelete/private.h]: Done.

**Comment by tklengyel** [comment]
Thanks for cleaning it up :) This is in much better shape now.

**Pull request merged**